roofit/roofitcore/src/RooPlot.cxx

```diff
@@ -100,16 +98,10 @@ RooPlot::RooPlot(Double_t xmin, Double_t xmax) :
   _hist(0), _items(), _plotVarClone(0), _plotVarSet(0), _normObj(0),
   _defYmin(1e-5), _defYmax(1), _dir(0)
 {
-  Bool_t histAddDirStatus = TH1::AddDirectoryStatus();
-  TH1::AddDirectory(kFALSE) ;
-
   _hist = new TH1D(histName(),"A RooPlot",100,xmin,xmax) ;
   _hist->Sumw2(kFALSE) ;
   _hist->GetSumw2()->Set(0) ;
-
-
-  TH1::AddDirectory(histAddDirStatus) ;
-
+  _hist->SetDirectory(nullptr);
```

Comment:
is 

```c++
_hist = new TH1D(histName(),"A RooPlot",100,xmin,xmax);
_hist->SetDirectory(nullptr);
```

strictly equivalent to

```c++
TH1::AddDirectory(kFALSE);
_hist = new TH1D(histName(),"A RooPlot",100,xmin,xmax);
```
?

There might be some global state change in the first version that is skipped in the second, not sure if it can have user-visible consequences... @pcanal ?

Reply:
Maybe not strictly equivalent, but I surely want to mess with global state as few as possible. According to docs, (and also what I always did as a user), setting the directory to `nullptr` gets you out of all the global state mess.